Objective-C/CBLDocument.mm

```diff
@@ -228,6 +228,11 @@ - (FLSliceResult) encode: (NSError**)outError {
     FLSliceResult body = FLEncoder_Finish(encoder, &flErr);
     if (!body.buf)
         createError(flErr, [NSString stringWithUTF8String: errMessage], outError);
+    
```

Comment:
Two things in this method:

1. The FLEncoderContext is missing a pointer of outHasAttachment. I think it could be like:

bool hasAttachment = false;
FLEncoderContext ctx = { ... , outHasAttachment = &hasAttachment }
```
2. Why is encodeQueryParameter true here? I might have misunderstood something.

Reply:
`encodeQueryParameter` probably a mistake in setting in CBLDocument. I have removed and tried running unit tests.

PR doesn't prove its required. 
https://github.com/couchbase/couchbase-lite-ios/pull/2812